doc/source/tutorial/arpack.rst

```diff
@@ -94,10 +94,10 @@ simplicity, we'll construct a symmetric, positive-definite matrix.
     >>> from scipy.linalg import eig, eigh
     >>> from scipy.sparse.linalg import eigs, eigsh
     >>> np.set_printoptions(suppress=True)
+    >>> rng = np.random.default_rng()
```

Comment:
Can we get reproducible results by seeding the `default_rng`?

Reply:
Yes but see https://github.com/scipy/scipy/pull/13863#issuecomment-820575709.

Reply:
And I just found back https://github.com/scipy/scipy/pull/13631#discussion_r600307550.

Reply:
I don't buy the idea that someone would copy code with random inputs. Surely they would use their own inputs if they're actually trying to achieve something other than learning the scipy function (in which case random quality is irrelevant).

However if it's really an issue, you could always use `default_rng(seed=SEED)` where `SEED` is defined in the doctest runner, so not part of the example. 
https://github.com/scipy/scipy/blob/374e5bfec63e530c34514a08d9311352138182f3/tools/refguide_check.py#L468-L469

Reply:
> However if it's really an issue, you could always use `default_rng(seed=SEED)` where `SEED` is defined in the doctest runner, so not part of the example.

Hmm, that would be a pleasing middle ground.

Reply:
Oh indeed, good point. I didn't realize this was possible. Then if it's okay I would do the change.

Reply:
Removing the seeding in all examples and using `# random` to make the doctest runner happy seems fine to me. `default_rng(seed=SEED)` without defining `SEED` would make code no longer run when copy-pasting, which is too high a price to pay imho.

Reply:
Ideally I think if you ran `make clean; make html` twice you'd get the same output, and with this PR you won't (better for reproducibility, smaller diffs with doc uploads to version controlled things, etc.). At the same time, I see the point about wanting to set a good example (which I agree is more important).

Another option is for for us to hack the doc building in some way to give us reproducible outputs. For example, in `conf.py` we could temporarily monkeypatch `numpy.random.default_rng` to be `lambda seed=0: original_default_rng(seed)`. That way our examples stay static, but readers just see us use `rng = numpy.random.default_rng()` in examples. I can look into this and open a PR to do this separately if people agree it solves our problems.

Reply:
Ark just seeing this. I started to add the SEED 😅. Well it's not for nothing if we take your approach @larsoner as I basically would have the new values for the tests... If we do this, you could commit to my branch directly. Or first we would use `# random` and then introduce this hack?

I will wait a bit for the decision then (+1 for @rgommers with the `# random`. In the end, I believe it could mean less technical debt  as we would not need to maintain changing values there.).

Just let me know 😃

Reply:
> If we do this, you could commit to my branch directly. 

This seems like the easiest approach. Let's wait to see if others agree it's the best compromise.

Also I do agree with @rgommers that we will cause copy-paste problems for people with the SEED approach, so I prefer the hack-the-doc-build approach.

Reply:
+1 for hack-the-doc-build, good idea

Reply:
ok great 😅 @larsoner I will push very soon what I did with `SEED`. Basically I added this in the `refguide_check.py`

```python
# SEED is generated with
# from numpy.random import SeedSequence
# print(SeedSequence().entropy)
DEFAULT_NAMESPACE = {'np': np, 'SEED': 1638083107694713882823079058616272161}
```
Then if I understand correctly your trick, please use this value when replacing `>>> rng = np.random.default_rng()`.